qiskit/ignis/verification/entanglement/parallelize.py

```diff
@@ -0,0 +1,449 @@
+then a normal "linear" CNOT gate preparation of the
+GHZ State. Additionally, there are methods within parallelize.py
+that configure different characterization tests for the
+GHZ State, including Multiple Quantum Coherence (MQC),
+Parity Oscillations (PO), and Quantum Tomography.
+'''
+
+from qiskit import *
+from qiskit.circuit import Parameter
+
+
+class BConfig:
+    '''
+    This class is used to create a GHZ circuit
+    with parallellized CNOT gates to increase fidelity
+    '''
+
+    def __init__(self, backend, indicator=True):
+        self.nodes = {}
+        self.backend = backend
+        self.cmap = backend.configuration().coupling_map
+        self.add_nodes()
+        self.indicator = indicator
+
+    def add_nodes(self):
+        '''
+        Adds nodes to the dictionary based on coupling map
+        '''
+        self.nodes = {}
+        for i in range(len(self.cmap_calib())):
+            self.nodes[self.cmap_calib()[i][0]] = []
+        for i in range(len(self.cmap_calib())):
+            self.nodes[self.cmap_calib()[i][0]].append(self.cmap_calib()[i][1])
+
+    def cmap_calib(self):
+        '''
+        Only intended for public devices (doubling and reversing
+        each item in coupling map), but useful to run anyway
+        '''
+        cmap_new = list(self.cmap)
+        for a in self.cmap:
+            if [a[1], a[0]] not in self.cmap:
+                cmap_new.append([a[1], a[0]])
+        cmap_new.sort(key=lambda x: x[0])
+        return cmap_new
+
+    def get_best_node(self):
+        '''
+        First node with the most connections; Does not yet sort
+        based on error, but that is probably not too useful
+        '''
+
+        bestNode = 0
+        for i in self.nodes:
+            if len(self.nodes[i]) > len(self.nodes[bestNode]):
+                bestNode = i
+            else:
+                pass
+        return bestNode
+
+    def indicator_off(self):
+        '''
+        We turn off gate-based sorting of the tier_dict
+        '''
+
+        self.indicator = False
+
+    def indicator_on(self):
+        '''
+        We turn off gate-based sorting of the tier_dict
+        '''
+
+        self.indicator = True
+
+    def get_cx_error(self):
+        '''
+        Gets dict of relevant CNOT gate errors
+        '''
+
+        a = self.backend.properties().to_dict()['gates']
+        cxerrordict = {}
+        for i in a:
+            if len(i['qubits']) == 1:
+                continue
+            if len(i['qubits']) == 2:
+                b = tuple(i['qubits'])
+                if b in cxerrordict.keys():
+                    pass
+                cxerrordict[b] = i['parameters'][0]['value']
+                if (b[1], b[0]) not in cxerrordict.keys():
+                    cxerrordict[(b[1], b[0])] = i['parameters'][0]['value']
+                else:
+                    pass
+
+        newcxerrordict = cxerrordict
+
+#         for a in list(cxerrordict.keys()):
```

Comment:
It's probably best not to keep commented-out code in the version that is to be merged.

Reply:
done (everywhere)